apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,260 @@
+import { useMemo } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { Combobox, ComboboxOption } from "@leafygreen-ui/combobox";
+import { BasicEmptyState } from "@leafygreen-ui/empty-state";
+import { palette } from "@leafygreen-ui/palette";
+import { ListSkeleton } from "@leafygreen-ui/skeleton-loader";
+import { H3, Body, H3Props, Disclaimer } from "@leafygreen-ui/typography";
+import pluralize from "pluralize";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useVersionAnalytics } from "analytics";
+import TextInputWithValidation from "components/TextInputWithValidation";
+import { failedTaskStatuses, taskStatusToCopy } from "constants/task";
+import { useToastContext } from "context/toast";
+import {
+  TestAnalysisQuery,
+  TestAnalysisQueryVariables,
+} from "gql/generated/types";
+import { TEST_ANALYSIS } from "gql/queries";
+import { useQueryParam, useQueryParams } from "hooks/useQueryParam";
+import { TestAnalysisQueryParams } from "types/task";
+import { reportError } from "utils/errorReporting";
+import { validateRegexp } from "utils/validators";
+import GroupedTestMapList from "./GroupedTestMapList";
+import { TaskBuildVariantField } from "./types";
+import {
+  countTotalTests,
+  filterGroupedTests,
+  getAllBuildVariants,
+  getAllTaskStatuses,
+  groupTestsByName,
+} from "./utils";
+
+const { green } = palette;
+interface TestAnalysisProps {
+  versionId: string;
+}
+const TestAnalysis: React.FC<TestAnalysisProps> = ({ versionId }) => {
+  const [selectedTaskStatuses, setSelectedTaskStatuses] = useQueryParam<
+    string[]
+  >(TestAnalysisQueryParams.Statuses, []);
+
+  const [selectedBuildVariants, setSelectedBuildVariants] = useQueryParam<
+    string[]
+  >(TestAnalysisQueryParams.Variants, []);
+
+  const [testName, setTestName] = useQueryParam(
```

Comment:
nit. when submitting with an empty string, it ends up in the URL with `testName=`. I think it should maybe be omitted